tests/unit/src/main/scala/scala/meta/internal/metals/debug/TestDebugger.scala

```diff
@@ -85,11 +85,12 @@ final class TestDebugger(
 
   def restart: Future[Unit] = {
     Debug.printEnclosing()
-    ifNotFailed(debugger.restart).andThen { case _ =>
-      debugger = connect(this)
-      terminated = Promise()
-      output = new DebuggeeOutput
-      breakpoints = new DebuggeeBreakpoints
+    ifNotFailed(debugger.restart).flatMap(_ => terminated.future).andThen {
```

Comment:
Just a minor change, I don't think it was causing any errors, but we would start new server before terminated the previous one.